wandb/sdk/launch/runner/gcp_vertex.py

```diff
@@ -147,27 +148,33 @@ def run(self, launch_project: LaunchProject) -> Optional[AbstractRun]:
             )
 
             generate_docker_image(
-                self._api,
                 launch_project,
                 image_uri,
                 entry_point,
                 docker_args,
                 runner_type="gcp-vertex",
             )
 
-        # todo: we don't always want to push the image (eg if image already hosted on gcp), figure this out
-        repo, tag = image_uri.split(":")
-        docker.push(repo, tag)
+        image, tag = image_uri.split(":")
+        if not exists_on_gcp(image, tag):
+            docker.push(image, tag)
 
         if not self.ack_run_queue_item(launch_project):
             return None
 
-        env_dict = get_env_vars_dict(launch_project, self._api)
+        # todo: this is super broken
+        env_prefix = [
+            "-e {}={}".format(k, v)
+            for k, v in get_env_vars_dict(launch_project, self._api).items()
+        ]
+        entry_cmd = get_entry_point_command(entry_point, launch_project.override_args)[
+            0
+        ].split()
 
         job = aiplatform.CustomContainerTrainingJob(
             display_name=gcp_training_job_name,
             container_uri=image_uri,
-            model_serving_container_environment_variables=env_dict,
+            command=env_prefix + entry_cmd,
```

Comment:
does this work?

Reply:
nope lol comment above has a todo saying it's broken